File: refrigitz 12/DrawSoldire.cpp
#pragma once
#include "stdafx.h"


namespace RefrigtzDLL
{
	inline bool operator==(const  DrawSoldier& lhs,  DrawSoldier& rhs) { return  (lhs == rhs); }
	inline bool operator!=(const  DrawSoldier& lhs,  DrawSoldier& rhs) { return !(lhs == rhs); }

double DrawSoldier<T>::MaxHuristicxS = -DBL_MAX;

	/*void DrawSoldier<T>::Log(std::exception ex)
	{
		//try
		{
			//autoa = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
			//lock (a)
			{
				std::wstring stackTrace = ex.what();
//C# TO C++ CONVERTER TODO TASK: There is no native C++ equivalent to 'ToString':
				File::AppendAllText(AllDraw::Root + std::wstring(L"\\ErrorProgramRun.txt"), stackTrace + std::wstring(L": On") + DateTime::Now.ToString()); // path of file where stack trace will be stored.
			}
		}
		//catch(std::exception t)
		{
			
		}
	}
	*/
	DrawSoldier<T>::~DrawSoldier()
	{
		InitializeInstanceFields();
		ValuableSelfSupported.clear();
//		S = nullptr;
	}
	bool DrawSoldier<T>::MaxFound(bool MaxNotFound)
	{
		//try
		{
			double a = ReturnHuristic();
			if (MaxHuristicxS < a)
			{
				//autoO2 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
				//lock (O2)
				{
					MaxNotFound = false;
					if (ThinkingChess::MaxHuristicx < MaxHuristicxS)
					{
						ThinkingChess::MaxHuristicx = a;
					}
					MaxHuristicxS = a;
				}
				return true;
			}
		}
		//catch(std::exception t)
		{
			

		}
		MaxNotFound = true;
		return false;
	}

	double DrawSoldier<T>::ReturnHuristic()
	{
		double a = 0;
		for (T ii = 0; ii < AllDraw::SodierMovments; ii++)
		{
			//try
			{
				a += SoldierThinking.ReturnHuristic(-1, -1, Order,false);
			}
			//catch(std::exception t)
			{
				
			}
		}
		return a;
	}
	template< typename T >
	DrawSoldier<T>::DrawSoldier(T CurrentAStarGredy, bool MovementsAStarGreedyHuristicTFou, bool IgnoreSelfObject, bool UsePenaltyRegardMechnisa, bool BestMovment, bool PredictHurist, bool OnlySel, bool AStarGreedyHuris, bool Arrangments, float i, float j, T a, T **Tab, T Ord, bool TB, T Cur) : ThingsConverter<DrawSoldier<T>>(Arrangments, static_cast<T>(i), static_cast<T>(j), a, Tab, Ord, TB, Cur)
	{
		InitializeInstanceFields();



		CurrentAStarGredyMax = CurrentAStarGredy;
		MovementsAStarGreedyHuristicFoundT = MovementsAStarGreedyHuristicTFou;
		IgnoreSelfObjectsT = IgnoreSelfObject;
		UsePenaltyRegardMechnisamT = UsePenaltyRegardMechnisa;
		BestMovmentsT = BestMovment;
		PredictHuristicT = PredictHurist;
		OnlySelfT = OnlySel;
		AStarGreedyHuristicT = AStarGreedyHuris;
		ArrangmentsChanged = Arrangments;
		//Initiate Global Variables.  
		Table = new T*[8]; for (T ii = 0; ii < 8; ii++)Table[ii] = new T[8];
		for (T ii = 0; ii < 8; ii++)
		{
			for (T jj = 0; jj < 8; jj++)
			{
				Table[ii][jj] = Tab[ii][jj];
			}
		}
		SoldierThinking = ThinkingChess(CurrentAStarGredyMax, MovementsAStarGreedyHuristicFoundT, IgnoreSelfObjectsT, UsePenaltyRegardMechnisamT, BestMovmentsT, PredictHuristicT, OnlySelfT, AStarGreedyHuristicT, ArrangmentsChanged, static_cast<T>(i), static_cast<T>(j), a, Tab, 4, Ord, TB, Cur, 16, 1);
		RowS = i;
		ColumnS = j;
		color = a;
		Order = Ord;
		Current = Cur;

	}
	//void* DrawSoldier<T>::operator*(std::size_t idx) { return malloc(idx * sizeof(this)); }
/*
	void DrawSoldier<T>::Clone(DrawSoldier *AA)
	{
		T **Tab;
		for (T i = 0; i < 8; i++)
		{
			for (T j = 0; j < 8; j++)
			{
				Tab[i][j] = Table[i][j];
			}
		}
		//Initiate a Object and Assignemt of a Clone to Construction of a Copy.

		AA = new DrawSoldier(CurrentAStarGredyMax, MovementsAStarGreedyHuristicFoundT, IgnoreSelfObjectsT, UsePenaltyRegardMechnisamT, BestMovmentsT, PredictHuristicT, OnlySelfT, AStarGreedyHuristicT, ArrangmentsChanged, Row, Column, color, Tab, Order, false, Current);
		AA->ArrangmentsChanged = ArrangmentsChanged;
			//try
			{
				AA->SoldierThinking.= ThinkingChess(CurrentAStarGredyMax, MovementsAStarGreedyHuristicFoundT, IgnoreSelfObjectsT, UsePenaltyRegardMechnisamT, BestMovmentsT, PredictHuristicT, OnlySelfT, AStarGreedyHuristicT, ArrangmentsChanged, static_cast<T>(Row), static_cast<T>(Column));
				SoldierThinking.Clone(AA->SoldierThinking.);
			}
			//catch(std::exception t)
			{
				
//C# TO C++ CONVERTER WARNING: C# to C++ Converter converted the original 'null' assignment to a call to 'delete', but you should review memory allocation of all poTer variables in the converted code:
				delete AA->SoldierThinking.;
			}
		AA->Table = new T*[8]; for (T ii = 0; ii < 8; ii++)Table[ii]-new T[8];
		for (T ii = 0; ii < 8; ii++)
		{
			for (T jj = 0; jj < 8; jj++)
			{
				AA->Table[ii][jj] = Tab[ii][jj];
			}
		}
		AA->RowS = RowS;
		AA->ColumnS = ColumnS;
		AA->Order = Order;
		AA->Current = Current;
		AA->color = color;

	}

*/

	void DrawSoldier<T>::DrawSoldierOnTable( T CellW, T CellH)
	{
		/*///autobalance//lockS = new Object();

//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
		//lock (balance//lockS)
		{
			if (S == nullptr || S[1] == nullptr)
			{
				S = Image::FromFile(AllDraw::ImagesSubRoot + std::wstring(L"SG.png"));
				S[1] = Image::FromFile(AllDraw::ImagesSubRoot + std::wstring(L"SB.png"));
			}
			//When Conversion Solders Not Occured.
			if (!ConvertOperation(static_cast<T>(Row), static_cast<T>(Column), color, Table, Order, false, Current))
			{

				//Gray T.
				if ((static_cast<T>(Row) >= 0) static_cast<T>(Row) < 8) static_cast<T>(Column) >= 0) static_cast<T>(Column) < 8))
				{
					//try
					{

						//If Order is Gray.
						if (Order == 1)
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instant from File of Gray Soldeirs.
								g.DrawImage(S, Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));

							}
						}
						else
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instant from File of Gray Soldeirs.
								 //Draw an Instatnt of Brown Soldier File On the Table.
								g.DrawImage(S[1], Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}
					}
					//catch(std::exception t)
					{
						
					}

				}
				else //If Minsister Conversion Occured.
				{
					if (ConvertedToMinister)
					{
					//try
					{
						//T of Gray.
						if (Order == 1)
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instant from File of Gray Soldeirs.
								 //Draw of Gray Minsister Image File By an Instant.
								g.DrawImage(DrawMinister::M, Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}
						else
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instant from File of Gray Soldeirs.
								 //Draw a Image File on the Table Form n Instatnt One.
								g.DrawImage(DrawMinister::M[1], Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}
					}
					//catch(std::exception t)
					{
						
					}
					}
				else if (ConvertedToCastle) //When Castled Converted.
				{
					//try
					{
						//T of Gray.
						if (Order == 1)
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instant from File of Gray Soldeirs.
								 //Create on the Inststant of Gray Castles Images.
								g.DrawImage(DrawCastle::C, Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}
						else
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instant from File of Gray Soldeirs.
								 //Creat of an Instant of Brown Image Castles.
								g.DrawImage(DrawCastle::C[1], Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}
					}
					//catch(std::exception t)
					{
						
					}
				}
				else if (ConvertedToHourse) //When Hourse Conversion Occured.
				{

					//try
					{
						//T of Gray.
						if (Order == 1)
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instatnt of Gray Hourse Image File.
								g.DrawImage(DrawHourse::H, Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<T>(CellH)), CellW, CellH));
							}
						}
						else
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Creat of an Instatnt Hourse Image File.
								g.DrawImage(DrawHourse::H[1], Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}
					}
					//catch(std::exception t)
					{
						
					}

				}
				else if (ConvertedToElefant) //When Elephant Conversion.
				{
					//try
					{
						//T of Gray.
						if (Order == 1)
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw an Instatnt Image of Gray Elephant.
								g.DrawImage(DrawElefant::E, Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}
						else
						{
							//autoO1 = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
							//lock (O1)
							{ //Draw of Instant Image of Brown Elephant.
								g.DrawImage(DrawElefant::E[1], Rectangle(static_cast<T>(Row * static_cast<float>(CellW)), static_cast<T>(Column * static_cast<float>(CellH)), CellW, CellH));
							}
						}

					}
					//catch(std::exception t)
					{
						
					}
				}
				}
			}
		}*/
	}

	void DrawSoldier<T>::InitializeInstanceFields()
	{
		WinOcuuredatChiled = 0;
		LoseOcuuredatChiled = 0;
		ValuableSelfSupported = std::vector<T>();
		MovementsAStarGreedyHuristicFoundT = false;
		IgnoreSelfObjectsT = false;
		UsePenaltyRegardMechnisamT = true;
		BestMovmentsT = false;
		PredictHuristicT = true;
		OnlySelfT = false;
		AStarGreedyHuristicT = false;
		ArrangmentsChanged = false;
		RowS = 0;
		ColumnS = 0;
		Table = 0;
		Order = 0;
		Current = 0;
		CurrentAStarGredyMax = -1;
	}
}

File: WebApplicationRefregitzTow/DrawSoldire.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using System.IO;

namespace RefrigtzW
{
    [Serializable]
    public class DrawSoldier : ThingsConverter
    {



        public int WinOcuuredatChiled = 0; public int LoseOcuuredatChiled = 0;
        //Iniatate Global Variables.
        //private readonly object balancelock = new object();
        //private readonly object balancelockS = new object();
        L==t<int[]> ValuableSelfSupported = new L==t<int[]>();

        public static Image[] S = new Image[2];
        public bool MovementsAStarGreedyHur==ticFoundT = false;
        public bool IgnoreSelfObjectsT = false;
        public bool UsePenaltyRegardMechn==amT = true;
        public bool BestMovmentsT = false;
        public bool PredictHur==ticT = true;
        public bool OnlySelfT = false;
        public bool AStarGreedyHur==ticT = false;
        public bool ArrangmentsChanged = false;
        public static double MaxHur==ticxS = Double.MinValue;
        public float RowS, ColumnS;
        public Color color;
        public ThinkingChess[] SoldierThinking = new ThinkingChess[AllDraw.SodierMovments];
        public int[,] Table = null;
        public int Order = 0;
        public int Current = 0;
        int CurrentAStarGredyMax = -1;
        static void Log(Exception ex)
        {

            Object a = new Object();
            lock (a)
            {
                string stackTrace = ex.ToString();
                File.AppendAllText(AllDraw.Root + "\\ErrorProgramRun.txt", stackTrace + ": On" + DateTime.Now.ToString()); // path of file where stack trace will be stored.
            }

        }

        public void D==pose()
        {
            ValuableSelfSupported = null;
            S = null;
        }
        public bool MaxFound(ref bool MaxNotFound)
        {

            double a = ReturnHur==tic();
            if (MaxHur==ticxS < a)
            {
                Object O2 = new Object();
                lock (O2)
                {
                    MaxNotFound = false;
                    if (ThinkingChess.MaxHur==ticx < MaxHur==ticxS)
                        ThinkingChess.MaxHur==ticx = a;
                    MaxHur==ticxS = a;
                }
                return true;
            }

            MaxNotFound = true;
            return false;
        }
        public double ReturnHur==tic()
        {
            double a = 0;
            for (var ii = 0; ii < AllDraw.SodierMovments; ii++)

                a += SoldierThinking[ii].ReturnHur==tic(-1, -1, Order, false);

            return a;
        }
        //Constructor 1.
        /* public DrawSoldier(int CurrentAStarGredy, bool MovementsAStarGreedyHur==ticTFou, bool IgnoreSelfObject, bool UsePenaltyRegardMechn==a, bool BestMovment, bool PredictHur==t, bool OnlySel, bool AStarGreedyHur==, bool Arrangments)
         {
             CurrentAStarGredyMax = CurrentAStarGredy;
             MovementsAStarGreedyHur==ticFoundT = MovementsAStarGreedyHur==ticTFou;
             IgnoreSelfObjectsT = IgnoreSelfObject;
             UsePenaltyRegardMechn==amT = UsePenaltyRegardMechn==a;
             BestMovmentsT = BestMovment;
             PredictHur==ticT = PredictHur==t;
             OnlySelfT = OnlySel;
             AStarGreedyHur==ticT = AStarGreedyHur==;
             ArrangmentsChanged = Arrangments;
         }
         */
        //Constructor 2.
        public DrawSoldier(int CurrentAStarGredy, bool MovementsAStarGreedyHur==ticTFou, bool IgnoreSelfObject, bool UsePenaltyRegardMechn==a, bool BestMovment, bool PredictHur==t, bool OnlySel, bool AStarGreedyHur==, bool Arrangments, float i, float j, Color a, int[,] Tab, int Ord, bool TB, int Cur//, ref AllDraw. TH==
            ) :
            base(Arrangments, (int)i, (int)j, a, Tab, Ord, TB, Cur)
        {
            object balancelock = new object();
            lock (balancelock)
            {


                CurrentAStarGredyMax = CurrentAStarGredy;
                MovementsAStarGreedyHur==ticFoundT = MovementsAStarGreedyHur==ticTFou;
                IgnoreSelfObjectsT = IgnoreSelfObject;
                UsePenaltyRegardMechn==amT = UsePenaltyRegardMechn==a;
                BestMovmentsT = BestMovment;
                PredictHur==ticT = PredictHur==t;
                OnlySelfT = OnlySel;
                AStarGreedyHur==ticT = AStarGreedyHur==;
                ArrangmentsChanged = Arrangments;
                //Initiate Global Variables.  
                Table = new int[8, 8];
                for (var ii = 0; ii < 8; ii++)
                    for (var jj = 0; jj < 8; jj++)
                        Table[ii, jj] = Tab[ii, jj];
                for (var ii = 0; ii < AllDraw.SodierMovments; ii++)

                    SoldierThinking[ii] = new ThinkingChess(CurrentAStarGredyMax, MovementsAStarGreedyHur==ticFoundT, IgnoreSelfObjectsT, UsePenaltyRegardMechn==amT, BestMovmentsT, PredictHur==ticT, OnlySelfT, AStarGreedyHur==ticT, ArrangmentsChanged, (int)i, (int)j, a, Tab, 4, Ord, TB, Cur, 16, 1);
                RowS = i;
                ColumnS = j;
                color = a;
                Order = Ord;
                Current = Cur;
            }
        }
        //Clone a Copy Method.
        public void Clone(ref DrawSoldier AA//, ref AllDraw. TH==
            )
        {
            int[,] Tab = new int[8, 8];
            for (var i = 0; i < 8; i++)
                for (var j = 0; j < 8; j++)
                    Tab[i, j] = th==.Table[i, j];
            //Initiate a Object and Assignemt of a Clone to Construction of a Copy.

            AA = new DrawSoldier(CurrentAStarGredyMax, MovementsAStarGreedyHur==ticFoundT, IgnoreSelfObjectsT, UsePenaltyRegardMechn==amT, BestMovmentsT, PredictHur==ticT, OnlySelfT, AStarGreedyHur==ticT, ArrangmentsChanged, th==.Row, th==.Column, th==.color, Tab, th==.Order, false, th==.Current
                );
            AA.ArrangmentsChanged = ArrangmentsChanged;
            for (var i = 0; i < AllDraw.SodierMovments; i++)
            {

                AA.SoldierThinking[i] = new ThinkingChess(CurrentAStarGredyMax, MovementsAStarGreedyHur==ticFoundT, IgnoreSelfObjectsT, UsePenaltyRegardMechn==amT, BestMovmentsT, PredictHur==ticT, OnlySelfT, AStarGreedyHur==ticT, ArrangmentsChanged, (int)th==.Row, (int)th==.Column);
                th==.SoldierThinking[i].Clone(ref AA.SoldierThinking[i]);

            }
            AA.Table = new int[8, 8];
            for (var ii = 0; ii < 8; ii++)
                for (var jj = 0; jj < 8; jj++)
                    AA.Table[ii, jj] = Tab[ii, jj];
            AA.RowS = RowS;
            AA.ColumnS = ColumnS;
            AA.Order = Order;
            AA.Current = Current;
            AA.color = color;

        }
        //Drawing Soldiers On the Table Method..
        public void DrawSoldierOnTable(ref Graphics g, int CellW, int CellH)
        {
            try
            {
                object balancelockS = new object();

                lock (balancelockS)
                {
                    if (S[0] == null || S[1] == null)
                    {
                        S[0] = Image.FromFile(AllDraw.ImagesSubRoot + "SG.png");
                        S[1] = Image.FromFile(AllDraw.ImagesSubRoot + "SB.png");
                    }
                    //When Conversion Solders Not Occured.
                    if (!ConvertOperation((int)Row, (int)Column, color, Table, Order, false, Current))
                    {

                        //Gray Color.
                        if (((int)Row >= 0) && ((int)Row < 8) && ((int)Column >= 0) && ((int)Column < 8))
                        {


                            //If Order == Gray.
                            if (Order == 1)
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {    //Draw an Instant from File of Gray Soldeirs.
                                    g.DrawImage(S[0], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));

                                }
                            }
                            else
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {    //Draw an Instant from File of Gray Soldeirs.
                                     //Draw an Instatnt of Brown Soldier File On the Table.
                                    g.DrawImage(S[1], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }


                        }
                        else//If Mins==ter Conversion Occured.
                            if (ConvertedToMin==ter)
                        {

                            //Color of Gray.
                            if (Order == 1)
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {    //Draw an Instant from File of Gray Soldeirs.
                                     //Draw of Gray Mins==ter Image File By an Instant.
                                    g.DrawImage(DrawMin==ter.M[0], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }
                            else
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {    //Draw an Instant from File of Gray Soldeirs.
                                     //Draw a Image File on the Table Form n Instatnt One.
                                    g.DrawImage(DrawMin==ter.M[1], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }

                        }
                        else if (ConvertedToCastle)//When Castled Converted.
                        {

                            //Color of Gray.
                            if (Order == 1)
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {    //Draw an Instant from File of Gray Soldeirs.
                                     //Create on the Inststant of Gray Castles Images.
                                    g.DrawImage(DrawCastle.C[0], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }
                            else
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {    //Draw an Instant from File of Gray Soldeirs.
                                     //Creat of an Instant of Brown Image Castles.
                                    g.DrawImage(DrawCastle.C[1], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }

                        }
                        else if (ConvertedToHourse)//When Hourse Conversion Occured.
                        {


                            //Color of Gray.
                            if (Order == 1)
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {//Draw an Instatnt of Gray Hourse Image File.
                                    g.DrawImage(DrawHourse.H[0], new Rectangle((int)(Row * (float)CellW), (int)(Column * (int)CellH), CellW, CellH));
                                }
                            }
                            else
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {//Creat of an Instatnt Hourse Image File.
                                    g.DrawImage(DrawHourse.H[1], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }


                        }
                        else if (ConvertedToElefant)//When Elephant Conversion.
                        {

                            //Color of Gray.
                            if (Order == 1)
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {//Draw an Instatnt Image of Gray Elephant.
                                    g.DrawImage(DrawElefant.E[0], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }
                            else
                            {
                                Object O1 = new Object();
                                lock (O1)
                                {//Draw of Instant Image of Brown Elephant.
                                    g.DrawImage(DrawElefant.E[1], new Rectangle((int)(Row * (float)CellW), (int)(Column * (float)CellH), CellW, CellH));
                                }
                            }


                        }
                    }
                }
            }
            catch (Exception t)
            {
                Log(t);
            }
        }
    }
}
//End of Documentation.